x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/grouping/Categorize.java

```diff
@@ -92,6 +93,12 @@ public boolean foldable() {
         return false;
     }
 
+    @Override
+    public Nullability nullable() {
+        // Both nulls and empty strings result in null values
```

Comment:
++ for the comment, thanks!

Reply:
To be precisely: any string for which the analyzer returns zero tokens results in a null category. The analyzer discards tokens like numbers, hex.numbers, and stopwords like Jan, Feb, Mon, Tue, ...